docs/dvmcp-spec-2025-03-26.md

```diff
@@ -0,0 +1,833 @@
+| 21316 | Feedback/Notifications (Ephemeral)               |
+
+## Server Discovery
+DVMCP provides two methods of server discovery, the main differences between these two methods being the visibility of the servers and the way they are advertised. Public servers can advertise themselves and their capabilities to improve discoverability when providing a "public" or accessible service. Private servers may not advertise themselves and their capabilities, but they can be discovered by clients that know the provider's public key or server identifier.
+
+### Discovery via Server Announcements (Public Servers)
+
+Providers announce their servers and capabilities by publishing events with kinds 31316 (server), 31317 (tools/list), 31318 (resources/list), and 31319 (prompts/list).
+
+#### Server Announcement Event
+
+```json
+{
+  "kind": 31316,
+  "pubkey": "<provider-pubkey>",
+  "content": {
+    "jsonrpc": "2.0",
+    "id": 1,
+    "result": {
+      "protocolVersion": "2025-03-26",
+      "capabilities": {
+        "logging": {},
+        "prompts": {
+          "listChanged": true
+        },
+        "resources": {
+          "subscribe": true,
+          "listChanged": true
+        },
+        "tools": {
+          "listChanged": true
+        }
+      },
+      "serverInfo": {
+        "name": "ExampleServer",
+        "version": "1.0.0"
+      },
+      "instructions": "Optional instructions for the client"
+    }
+  },
+  "tags": [
+    ["d", "<server-identifier>"],          // Required: Unique identifier for the server
+    ["name", "Example Server"],            // Optional: Human-readable server name
+    ["about", "Server description"],       // Optional: Server description
+    ["picture", "https://example.com/server.png"],  // Optional: Server icon/avatar URL
+    ["website", "https://example.com"],    // Optional: Server website
+    ["k", "5910"],                         // Required: Accepted event kinds (for requests)
+    ["support_encryption", "true"]         // Optional: Whether server supports encrypted messages
+  ]
+}
+```
+
+#### Tools List Event
+```json
+{
+  "kind": 31317,
+  "pubkey": "<provider-pubkey>",
+  "content": {
+    "jsonrpc": "2.0",
+    "id": 1,
+    "result": {
+      "tools": [
+        {
+          "name": "get_weather",
+          "description": "Get current weather information for a location",
+          "inputSchema": {
+            "type": "object",
+            "properties": {
+              "location": {
+                "type": "string",
+                "description": "City name or zip code"
+              }
+            },
+            "required": ["location"]
+          }
+        }
+      ]
+    }
+  },
+  "tags": [
+    ["d", "<unique-identifier>"],        // Required: Unique identifier for the tools list
+    ["s", "<server-identifier>"],        // Required: Reference to the server
+    ["t", "get_weather"]                // Required: One t tag per tool name for enhanced discoverability
```

Comment:
```suggestion
    ["t", "get_weather"],                // Required: One t tag per tool name for enhanced discoverability
    ["t", "get_weather_future"] 
```
Would be good to show that there is supposed to be one `t` tag per tool

also whats the purpose of having indexable tool tags in the tools list event? will there be a need to lookup tools named the same across multiple MCP servers?